src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        removeCarDrivers(car);
+        insertCarDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?"
+                + "AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement =
+                        connection.prepareStatement(deleteCarRequest)) {
+            deleteCarStatement.setObject(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a car with ID " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsQuery = "SELECT id, model, manufacturer_id FROM cars AS c "
+                + "INNER JOIN cars_drivers AS cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE is_deleted = FALSE AND driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement =
+                        connection.prepareStatement(getCarsQuery)) {
+            getDriversStatement.setObject(1, driverId);
+            ResultSet resultSet = getDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                Long carId = resultSet.getObject("id", Long.class);
+                get(carId).ifPresent(cars::add);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars by driver ID " + driverId, e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getDriversRequest = "SELECT * FROM cars_drivers cd "
+                + "JOIN drivers d ON d.id = cd.driver_id "
+                + "WHERE cd.car_id = ? AND d.is_deleted = FALSE;";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement =
+                        connection.prepareStatement(getDriversRequest)) {
+            getDriversStatement.setObject(1, carId);
+            ResultSet resultSet = getDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(createDriverEntity(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get drivers by car ID "
+                    + carId + " from DB", e);
+        }
+        return drivers;
+    }
+
+    private Manufacturer createManufacturerEntity(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
```
the same for other similar methods